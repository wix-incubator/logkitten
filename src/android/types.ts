import type { Entry, Filter } from '../types';

// Android-specific log entry
export interface AndroidEntry extends Entry {
  tag: string;
  uid: string;
}

// Android-specific options
export interface AndroidOptions {
  adbPath?: string;
  deviceId?: string;
}

export interface AliasedAndroidOptions extends AndroidOptions {
  platform: 'android';
  filter?: Filter<AndroidEntry>;
}
