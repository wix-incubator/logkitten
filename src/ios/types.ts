import type { Entry, Filter } from '../types';

// iOS-specific log entry
export interface IosEntry extends Entry {
  category: string;
  formatString: string;
  processImagePath: string;
  subsystem: string;
  userID: number;
}

// iOS-specific options
export interface IosOptions {
  deviceId?: string;
}

export interface AliasedIosOptions extends IosOptions {
  platform: 'ios';
  filter?: Filter<IosEntry>;
}

export interface RawEntryIOS {
  timezoneName: string;
  messageType: string;
  eventType: string;
  source: any;
  formatString: string;
  userID: number;
  activityIdentifier: number;
  subsystem: string;
  category: string;
  threadID: number;
  senderImageUUID: string;
  backtrace: RawEntryIOS$Backtrace;
  bootUUID: string;
  processImagePath: string;
  senderImagePath: string;
  timestamp: string;
  machTimestamp: number;
  eventMessage: string;
  processImageUUID: string;
  traceID: number;
  processID: number;
  senderProgramCounter: number;
  parentActivityIdentifier: number;
}

export interface RawEntryIOS$Backtrace {
  frames: RawEntryIOS$Frame[];
}

export interface RawEntryIOS$Frame {
  imageOffset: number;
  imageUUID: string;
}
