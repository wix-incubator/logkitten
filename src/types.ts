export type Platform = 'ios' | 'android';

export type Entry = {
  /** Timestamp in milliseconds */
  ts: number;
  /** Process ID */
  pid: number;
  /** Priority: 0 - unknown, 1 - debug, 2 - info, 3 - default, 4 - warning, 5 - error, 6 - fault */
  priority: number;
  /** Tag */
  tag?: string;
  /** App ID */
  appId?: string;
  /** Messages */
  messages: string[];
  /** Platform */
  platform: Platform;
  /** Process name */
  processName?: string;
  threadId?: number;
  subsystem?: string;
  category?: string;
};

export interface Parser {
  splitMessages(data: string): string[];
  parseMessages(messages: string[]): Entry[];
}

export type Filter = (entry: Entry) => boolean;
