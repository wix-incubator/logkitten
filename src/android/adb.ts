import { spawn, execFileSync, ChildProcess } from 'child_process';
import path from 'path';
import {
  CodeError,
  ERR_ANDROID_CANNOT_CLEAN_LOGCAT_BUFFER,
  ERR_ANDROID_CANNOT_START_LOGCAT,
} from '../errors';

export function runAndroidLoggingProcess(
  adbPath?: string,
  deviceId?: string
): ChildProcess {
  const execPath = getAdbPath(adbPath);
  return spawnLogcatProcess(execPath, deviceId);
}

export function getSdkRoot(): string | undefined {
  return process.env.ANDROID_SDK_ROOT ?? process.env.ANDROID_HOME;
}

export function getAdbPath(customPath?: string): string {
  if (customPath) {
    return path.resolve(customPath);
  }
  const sdkRoot = getSdkRoot();
  return sdkRoot ? `${sdkRoot}/platform-tools/adb` : 'adb';
}

export function spawnLogcatProcess(
  adbPath: string,
  deviceId?: string
): ChildProcess {
  const baseArgs = deviceId ? ['-s', deviceId] : [];

  try {
    execFileSync(adbPath, [...baseArgs, 'logcat', '-c']);
  } catch (error) {
    throw new CodeError(
      ERR_ANDROID_CANNOT_CLEAN_LOGCAT_BUFFER,
      (error as Error).message
    );
  }

  try {
    return spawn(
      adbPath,
      [...baseArgs, 'logcat', '-v', 'time', 'process', 'tag'],
      {
        stdio: 'pipe',
      }
    );
  } catch (error) {
    throw new CodeError(
      ERR_ANDROID_CANNOT_START_LOGCAT,
      (error as Error).message
    );
  }
}
