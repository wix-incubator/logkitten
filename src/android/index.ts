import type { PlatformFactory } from '../types';
import type { AndroidEntry, AndroidOptions } from './types';
import { run } from './runner';
import { AndroidParser } from './parser';

export const create: PlatformFactory<AndroidEntry, AndroidOptions> = (
  options
) => {
  return {
    name: 'android',
    process: run(options),
    parse: AndroidParser.create(),
  };
};

export type { AndroidOptions, AndroidEntry } from './types';
