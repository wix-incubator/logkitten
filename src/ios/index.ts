import type { PlatformFactory } from '../types';
import type { IosEntry, IosOptions } from './types';
import { run } from './runner';
import { IosParser } from './parser';

export const create: PlatformFactory<IosEntry, IosOptions> = (options) => ({
  name: 'ios',
  process: run(options),
  parse: IosParser.create(),
});

export { IosOptions, IosEntry } from './types';
