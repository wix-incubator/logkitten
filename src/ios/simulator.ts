import { ChildProcess, spawn } from 'child_process';
import { CodeError, ERR_IOS_CANNOT_START_SYSLOG } from '../errors';

export function runSimulatorLoggingProcess(deviceId?: string): ChildProcess {
  const targetDevice = deviceId || 'booted';

  try {
    return spawn(
      'xcrun',
      [
        'simctl',
        'spawn',
        targetDevice,
        'log',
        'stream',
        '--type',
        'log',
        '--level',
        'debug',
      ],
      {
        stdio: 'pipe',
      }
    );
  } catch (error) {
    throw new CodeError(ERR_IOS_CANNOT_START_SYSLOG, (error as Error).message);
  }
}
