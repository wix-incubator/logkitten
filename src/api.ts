/* Common */
import { EventEmitter } from 'events';
import { Entry, Platform } from './types';

/* Android */
import createAndroidFilter from './android/AndroidFilter';
import AndroidParser from './android/AndroidParser';
import { runAndroidLoggingProcess } from './android/adb';
export { Priority as AndroidPriority } from './android/constants';

/* iOS */
import IosParser from './ios/IosParser';
import createIosFilter from './ios/IosFilter';
import { runSimulatorLoggingProcess } from './ios/simulator';
import { CodeError, ERR_IOS_NO_SIMULATORS_BOOTED } from './errors';
export { Priority as IosPriority } from './ios/constants';

/* Exports */
export { Entry } from './types';

export type LogkittyOptions = {
  platform: Platform;
  adbPath?: string;
  priority?: number;
  filter?: (entry: Entry) => boolean;
};

export function logkitten(options: LogkittyOptions): EventEmitter {
  const { platform, adbPath, priority, filter: userFilter } = options;
  const emitter = new EventEmitter();

  if (
    !['ios', 'android'].some(
      (availablePlatform) => availablePlatform === platform
    )
  ) {
    throw new Error(`Platform ${platform} is not supported`);
  }

  const parser = platform === 'android' ? new AndroidParser() : new IosParser();
  const baseFilter =
    platform === 'android'
      ? createAndroidFilter(priority)
      : createIosFilter(priority);

  const loggingProcess =
    platform === 'android'
      ? runAndroidLoggingProcess(adbPath)
      : runSimulatorLoggingProcess();

  process.on('exit', () => {
    loggingProcess.kill();
    emitter.emit('exit');
  });

  loggingProcess.stderr?.on('data', (errorData: string | Buffer) => {
    if (platform === 'ios') {
      const msg = errorData.toString();

      if (msg.includes('getpwuid_r did not find a match for uid')) {
        return;
      }

      if (msg.includes('No devices are booted.')) {
        emitter.emit(
          'error',
          new CodeError(
            ERR_IOS_NO_SIMULATORS_BOOTED,
            'No simulators are booted.'
          )
        );
        return;
      }
    }

    emitter.emit('error', new Error(errorData.toString()));
  });

  loggingProcess.stdout?.on('data', (raw: string | Buffer) => {
    let entryToLog: Entry | undefined;
    try {
      const messages = parser.splitMessages(raw.toString());
      const entries = parser.parseMessages(messages);
      entries.forEach((entry: Entry) => {
        // Apply base priority filter and optional user filter
        if (baseFilter(entry) && (!userFilter || userFilter(entry))) {
          entryToLog = entry;
        }
      });
    } catch (error) {
      emitter.emit('error', error);
    }

    if (entryToLog) {
      emitter.emit('entry', entryToLog);
    }
  });

  loggingProcess.stdout?.on('error', (error: Error) => {
    emitter.emit('error', error);
    emitter.emit('exit');
  });

  return emitter;
}
