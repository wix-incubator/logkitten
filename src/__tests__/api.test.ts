import { EventEmitter } from 'events';
import { logkitten, AndroidPriority } from '../api';
import { runAndroidLoggingProcess } from '../android/adb';
import { runSimulatorLoggingProcess } from '../ios/simulator';
import { Entry } from '../types';
import { ANDROID_RAW_LOG_FIXTURES } from './__fixtures__/android';
import { IOS_RAW_LOG_FIXTURES } from './__fixtures__/ios';

jest.mock('../android/adb.ts', () => ({
  runAndroidLoggingProcess: jest.fn(),
}));

jest.mock('../ios/simulator.ts', () => ({
  runSimulatorLoggingProcess: jest.fn(),
}));

describe('Node API', () => {
  describe('for Android', () => {
    describe('should spawn logcat process and emit entries', () => {
      it('when no filter is used', (done: Function) => {
        let entriesEmitted = 0;
        const loggingEmitter = new EventEmitter();
        (runAndroidLoggingProcess as jest.Mock).mockImplementationOnce(() => ({
          stdout: loggingEmitter,
          stderr: new EventEmitter(),
        }));

        const emitter = logkitten({
          platform: 'android',
          priority: AndroidPriority.INFO,
        });

        emitter.on('entry', (entry: Entry) => {
          expect(entry).toMatchSnapshot();
          entriesEmitted += 1;
        });

        ANDROID_RAW_LOG_FIXTURES.forEach((data: string) => {
          loggingEmitter.emit('data', data);
        });

        setTimeout(() => {
          expect(entriesEmitted).toBe(4);
          done();
        });
      });

      it('when tag filter is used', (done: Function) => {
        let entriesEmitted = 0;
        const loggingEmitter = new EventEmitter();
        (runAndroidLoggingProcess as jest.Mock).mockImplementationOnce(() => ({
          stdout: loggingEmitter,
          stderr: new EventEmitter(),
        }));

        const emitter = logkitten({
          platform: 'android',
          priority: AndroidPriority.VERBOSE,
          filter: (entry) =>
            entry.tag === 'wificond' || entry.tag === 'storaged',
        });

        emitter.on('entry', (entry: Entry) => {
          expect(entry).toMatchSnapshot();
          entriesEmitted += 1;
        });

        ANDROID_RAW_LOG_FIXTURES.forEach((data: string) => {
          loggingEmitter.emit('data', data);
        });

        setTimeout(() => {
          expect(entriesEmitted).toBe(3);
          done();
        });
      });

      it('when message filter is used', (done: Function) => {
        let entriesEmitted = 0;
        const loggingEmitter = new EventEmitter();
        (runAndroidLoggingProcess as jest.Mock).mockImplementationOnce(() => ({
          stdout: loggingEmitter,
          stderr: new EventEmitter(),
        }));

        const emitter = logkitten({
          platform: 'android',
          priority: AndroidPriority.VERBOSE,
          filter: (entry) => entry.messages.some((msg) => /scan/.test(msg)),
        });

        emitter.on('entry', (entry: Entry) => {
          expect(entry).toMatchSnapshot();
          entriesEmitted += 1;
        });

        ANDROID_RAW_LOG_FIXTURES.forEach((data: string) => {
          loggingEmitter.emit('data', data);
        });

        setTimeout(() => {
          expect(entriesEmitted).toBe(2);
          done();
        });
      });

      it('should pass deviceId to Android logging process', () => {
        const loggingEmitter = new EventEmitter();
        (runAndroidLoggingProcess as jest.Mock).mockImplementationOnce(() => ({
          stdout: loggingEmitter,
          stderr: new EventEmitter(),
        }));

        logkitten({
          platform: 'android',
          deviceId: 'emulator-5554',
          priority: AndroidPriority.DEBUG,
        });

        expect(runAndroidLoggingProcess).toHaveBeenCalledWith(
          undefined,
          'emulator-5554'
        );
      });
    });
  });

  describe('for iOS', () => {
    describe('should spawn logging process and emit entires', () => {
      it('when no filter is used', (done: Function) => {
        let entriesEmitted = 0;
        const loggingEmitter = new EventEmitter();
        (runSimulatorLoggingProcess as jest.Mock).mockImplementationOnce(
          () => ({
            stdout: loggingEmitter,
            stderr: new EventEmitter(),
          })
        );

        const emitter = logkitten({
          platform: 'ios',
        });

        emitter.on('entry', (entry: Entry) => {
          expect(entry).toMatchSnapshot();
          entriesEmitted += 1;
        });

        IOS_RAW_LOG_FIXTURES.forEach((data: string) => {
          loggingEmitter.emit('data', data);
        });

        setTimeout(() => {
          expect(entriesEmitted).toBe(4);
          done();
        });
      });

      it('when tag filter is used', (done: Function) => {
        let entriesEmitted = 0;
        const loggingEmitter = new EventEmitter();
        (runSimulatorLoggingProcess as jest.Mock).mockImplementationOnce(
          () => ({
            stdout: loggingEmitter,
            stderr: new EventEmitter(),
          })
        );

        const emitter = logkitten({
          platform: 'ios',
          filter: (entry) => entry.tag === 'com.apple.network',
        });

        emitter.on('entry', (entry: Entry) => {
          expect(entry).toMatchSnapshot();
          if (entriesEmitted === 1) {
            expect(entry).toMatchSnapshot();
          }
          entriesEmitted += 1;
        });

        IOS_RAW_LOG_FIXTURES.forEach((data: string) => {
          loggingEmitter.emit('data', data);
        });

        setTimeout(() => {
          expect(entriesEmitted).toBe(3);
          done();
        }, 0);
      });

      it('when message filter is used', (done: Function) => {
        let entriesEmitted = 0;
        const loggingEmitter = new EventEmitter();
        (runSimulatorLoggingProcess as jest.Mock).mockImplementationOnce(
          () => ({
            stdout: loggingEmitter,
            stderr: new EventEmitter(),
          })
        );

        const emitter = logkitten({
          platform: 'ios',
          filter: (entry) =>
            entry.messages.some((msg) => /Socket SO_ERROR/.test(msg)),
        });

        emitter.on('entry', (entry: Entry) => {
          try {
            expect(entry).toMatchSnapshot();
            entriesEmitted += 1;
          } catch (error) {
            console.error('Test failed for entry:', entry, error);
            done(error);
          }
        });

        IOS_RAW_LOG_FIXTURES.forEach((data: string) => {
          loggingEmitter.emit('data', data);
        });

        setTimeout(() => {
          try {
            expect(entriesEmitted).toBe(1);
            done();
          } catch (error) {
            done(error);
          }
        }, 100);
      });

      it('should pass deviceId to iOS logging process', () => {
        const loggingEmitter = new EventEmitter();
        (runSimulatorLoggingProcess as jest.Mock).mockImplementationOnce(
          () => ({
            stdout: loggingEmitter,
            stderr: new EventEmitter(),
          })
        );

        logkitten({
          platform: 'ios',
          deviceId: 'A1B2C3D4-E5F6-7890-ABCD-EF1234567890',
        });

        expect(runSimulatorLoggingProcess).toHaveBeenCalledWith(
          'A1B2C3D4-E5F6-7890-ABCD-EF1234567890'
        );
      });
    });
  });

  it('should emit error when parsing fails', (done: Function) => {
    const loggingEmitter = new EventEmitter();
    (runAndroidLoggingProcess as jest.Mock).mockImplementationOnce(() => ({
      stdout: loggingEmitter,
      stderr: new EventEmitter(),
    }));

    const emitter = logkitten({
      platform: 'android',
    });

    emitter.on('error', (error: Error) => {
      expect(error).toBeDefined();
      done();
    });

    loggingEmitter.emit('data', null);
  });

  it('should throw error when unsupported platform is used', () => {
    expect(() => {
      logkitten({
        platform: 'windows' as any,
      });
    }).toThrow('Platform windows is not supported');
  });
});
