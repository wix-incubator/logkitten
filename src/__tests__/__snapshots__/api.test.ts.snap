// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Node API for Android should spawn logcat process and emit entries when message filter is used 1`] = `
{
  "messages": [
    "No pno scan started",
  ],
  "pid": 935,
  "platform": "android",
  "priority": 4,
  "tag": "wificond",
  "ts": 1767225599000,
}
`;

exports[`Node API for Android should spawn logcat process and emit entries when message filter is used 2`] = `
{
  "messages": [
    "Scheduled scan is not running!",
  ],
  "pid": 935,
  "platform": "android",
  "priority": 2,
  "tag": "wificond",
  "ts": 1767225600000,
}
`;

exports[`Node API for Android should spawn logcat process and emit entries when no filter is used 1`] = `
{
  "messages": [
    "getDiskStats failed with result NOT_SUPPORTED and size 0",
  ],
  "pid": 934,
  "platform": "android",
  "priority": 5,
  "tag": "storaged",
  "ts": 1767225593000,
}
`;

exports[`Node API for Android should spawn logcat process and emit entries when no filter is used 2`] = `
{
  "messages": [
    "uid=1000(system) ActivityManager expire 10 lines",
  ],
  "pid": 1383,
  "platform": "android",
  "priority": 3,
  "tag": "chatty",
  "ts": 1767225594000,
}
`;

exports[`Node API for Android should spawn logcat process and emit entries when no filter is used 3`] = `
{
  "messages": [
    "No pno scan started",
  ],
  "pid": 935,
  "platform": "android",
  "priority": 4,
  "tag": "wificond",
  "ts": 1767225599000,
}
`;

exports[`Node API for Android should spawn logcat process and emit entries when tag filter is used 1`] = `
{
  "messages": [
    "getDiskStats failed with result NOT_SUPPORTED and size 0",
  ],
  "pid": 934,
  "platform": "android",
  "priority": 5,
  "tag": "storaged",
  "ts": 1767225593000,
}
`;

exports[`Node API for Android should spawn logcat process and emit entries when tag filter is used 2`] = `
{
  "messages": [
    "No pno scan started",
  ],
  "pid": 935,
  "platform": "android",
  "priority": 4,
  "tag": "wificond",
  "ts": 1767225599000,
}
`;

exports[`Node API for Android should spawn logcat process and emit entries when tag filter is used 3`] = `
{
  "messages": [
    "Scheduled scan is not running!",
  ],
  "pid": 935,
  "platform": "android",
  "priority": 2,
  "tag": "wificond",
  "ts": 1767225600000,
}
`;

exports[`Node API for iOS should spawn logging process and emit entires when message filter is used 1`] = `
{
  "category": "connection",
  "messages": [
    "nw_socket_handle_socket_event [C3103.1.1:2] Socket SO_ERROR [61: Connection refused]",
  ],
  "pid": 87578,
  "platform": "ios",
  "priority": 5,
  "processName": "example",
  "subsystem": "com.apple.network",
  "tag": "com.apple.network",
  "threadId": 154652000,
  "ts": 1750765489382,
}
`;

exports[`Node API for iOS should spawn logging process and emit entires when no filter is used 1`] = `
{
  "category": "Summary",
  "messages": [
    "Task <96F5582A-8C99-4035-AEC1-29C47714BAD5>.<2408> summary for task failure {transaction_duration_ms=12, response_status=-1, connection=2408, reused=1, reused_after_ms=0, request_start_ms=0, request_duration_ms=0, response_start_ms=0, response_duration_ms=0, request_bytes=0, request_throughput_kbps=0, response_bytes=0, response_throughput_kbps=0, cache_hit=false}",
  ],
  "pid": 87578,
  "platform": "ios",
  "priority": 3,
  "processName": "example",
  "subsystem": "com.apple.CFNetwork",
  "tag": "com.apple.CFNetwork",
  "threadId": 154653588,
  "ts": 1750764754176,
}
`;

exports[`Node API for iOS should spawn logging process and emit entires when no filter is used 2`] = `
{
  "category": "connection",
  "messages": [
    "[C2430 Hostname#65c1aeb9:63995 tcp, url: http://localhost:63995/, definite, attribution: developer] dealloc",
  ],
  "pid": 87578,
  "platform": "ios",
  "priority": 2,
  "processName": "example",
  "subsystem": "com.apple.network",
  "tag": "com.apple.network",
  "threadId": 154652000,
  "ts": 1750764777470,
}
`;

exports[`Node API for iOS should spawn logging process and emit entires when no filter is used 3`] = `
{
  "category": "connection",
  "messages": [
    "nw_socket_handle_socket_event [C3103.1.1:2] Socket SO_ERROR [61: Connection refused]",
  ],
  "pid": 87578,
  "platform": "ios",
  "priority": 5,
  "processName": "example",
  "subsystem": "com.apple.network",
  "tag": "com.apple.network",
  "threadId": 154652000,
  "ts": 1750765489382,
}
`;

exports[`Node API for iOS should spawn logging process and emit entires when no filter is used 4`] = `
{
  "category": "activity",
  "messages": [
    "No threshold for cfnetwork:foreground_task, returning NW_ACTIVITY_DURATION_INVALID",
  ],
  "pid": 87578,
  "platform": "ios",
  "priority": 1,
  "processName": "example",
  "subsystem": "com.apple.network",
  "tag": "com.apple.network",
  "threadId": 154652005,
  "ts": 1750764801817,
}
`;

exports[`Node API for iOS should spawn logging process and emit entires when tag filter is used 1`] = `
{
  "category": "connection",
  "messages": [
    "[C2430 Hostname#65c1aeb9:63995 tcp, url: http://localhost:63995/, definite, attribution: developer] dealloc",
  ],
  "pid": 87578,
  "platform": "ios",
  "priority": 2,
  "processName": "example",
  "subsystem": "com.apple.network",
  "tag": "com.apple.network",
  "threadId": 154652000,
  "ts": 1750764777470,
}
`;

exports[`Node API for iOS should spawn logging process and emit entires when tag filter is used 2`] = `
{
  "category": "connection",
  "messages": [
    "nw_socket_handle_socket_event [C3103.1.1:2] Socket SO_ERROR [61: Connection refused]",
  ],
  "pid": 87578,
  "platform": "ios",
  "priority": 5,
  "processName": "example",
  "subsystem": "com.apple.network",
  "tag": "com.apple.network",
  "threadId": 154652000,
  "ts": 1750765489382,
}
`;

exports[`Node API for iOS should spawn logging process and emit entires when tag filter is used 3`] = `
{
  "category": "connection",
  "messages": [
    "nw_socket_handle_socket_event [C3103.1.1:2] Socket SO_ERROR [61: Connection refused]",
  ],
  "pid": 87578,
  "platform": "ios",
  "priority": 5,
  "processName": "example",
  "subsystem": "com.apple.network",
  "tag": "com.apple.network",
  "threadId": 154652000,
  "ts": 1750765489382,
}
`;

exports[`Node API for iOS should spawn logging process and emit entires when tag filter is used 4`] = `
{
  "category": "activity",
  "messages": [
    "No threshold for cfnetwork:foreground_task, returning NW_ACTIVITY_DURATION_INVALID",
  ],
  "pid": 87578,
  "platform": "ios",
  "priority": 1,
  "processName": "example",
  "subsystem": "com.apple.network",
  "tag": "com.apple.network",
  "threadId": 154652005,
  "ts": 1750764801817,
}
`;
